### PR TITLE
Vore mobs can eat you while unconcious

### PR DESCRIPTION
**Pull request by Very-Soft**
-Only if it's enabled on the mob's AI holder
-Only if your settings are configured to allow it.

Otherwise it acts normal, doesn't bother SSD people, doesn't maul you while unconscious (unless it is set to do that, since that's an option that exists)

Important for a mob I will be adding soon.

### CODE REVIEW

File: code/modules/ai/ai_holder_targeting.dm [hunk 1]
```diff
@@ -129,6 +130,8 @@
 			if(L.stat == UNCONSCIOUS)	// Do we have mauling? Yes? Then maul people who are sleeping but not SSD
 				if(mauling)
 					return TRUE
+				else if(unconscious_vore && L.allowmobvore) //VOREStation Add
```
Comment (Screemonster) [inline code/modules/ai/ai_holder_targeting.dm:133]: Could this create an edge case where the mob will eat unconscious people, the prey allows mob vore, but something else prevents the mob from eating (such as being full) so it reverts to the standard mauling behaviour and claws them to death?

Comment (Very-Soft) [reply code/modules/ai/ai_holder_targeting.dm:133]: That seems possible. I also was gonna look into what OOC escape does to make sure they don't maul people after they do that too.

Comment (Screemonster) [reply code/modules/ai/ai_holder_targeting.dm:133]: maybe return false if will_eat() fails or something?

**Comment by Razgriz1032** [comment]
What was the point of this PR? To globally disable mobs being able to eat unconscious people by changing the flag from TRUE to FALSE?

**Comment by Very-Soft** [comment]
> What was the point of this PR? To globally disable mobs being able to eat unconscious people by changing the flag from TRUE to FALSE?

By default mobs can't really do anything to unconscious people unless they started doing it before they became unconscious, unless you turn mauling on, on their AI holder, which allows them to just continue to attack until the person is dead. 

This uses the same argument as mauling, but only allows them to do anything if the person's vore settings allow them to be eaten by the mob in question. (which, when an AI controlled mob tries to attack something that's fallen down, it eats them if it can)

This was added to support enemy types that could knock you out without killing you and still allow for mob vore. (Such as the space jellyfish, which primarily deal halloss damage, which usually results in unconsciousness before knocking you over.)

**Pull request merged**